code/modules/mob/living/simple_animal/hostile/terror_spiders/ghost.dm

```diff
@@ -42,5 +42,6 @@
 		to_chat(user, "<span class='notice'>Someone else already took this spider.</span>")
 		return
 	key = user.key
+	walk(src, 0) // if spider is moving under ai control (already aggroed on something before player control), then stop AI movement
```

Comment:
Wouldn't it be better to do this in the spider's login proc? That gets called whenever a player takes over the mob regardless of source, which means it would also apply if, for example, an admin manually drags a ghost into the spider.

Reply:
Moved to mob/living/Login() so it handles that case, and also handles cases for other mobs (like headcrabs).